### PR TITLE
Create ROADMAP.md

### PR DESCRIPTION
**Pull request by stroomdev66**

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+## Reference data storage
+Reference data uses a memory mapped disk based store rather than direct memory to reduce the memory overhead associated with storing reference data. Reference data is also enhanced to cope with changes (additions and removals) of state information rather than always relying on complete snapshots.
+
+## Proxy processing
+Stroom proxy is capable of pipeline processing in the same way as a full Stroom application. Pipeline configuration content can be pushed to proxies so that they can perform local processing prior to sending data to Stroom. 
+
+## Multiple input sources
+Stroom is capable of processing data from a Kafka topic, HDFS, the local file system, HTTP POST in addition to the included stream store.
+
+In addition to processing data from multiple sources Stroom now has improved support for writing to various destinations.
```
Comment (gcdev373) [inline ROADMAP.md]: Don't understand the point of this sentence and find it a bit confusing.

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## Improved field extraction
+Enhancements to data splitter and associated UI to make the process of extracting field data from raw content much easier.
+
+## Kafka backed processing
```
Comment (gcdev373) [inline ROADMAP.md]: This isn't a feature.  If it is necessary to rewrite processing, then explain limitations of current approach before describing a solution.  In particular, say what will be better about the product when this is done - having Kafka in the mix will bring its own issues, e.g. deployment complexity, and I'm yet to be convinced that this is the answer.  

For this roadmap, perhaps just describe the task/feature as "Processing improvements (possibly using Kafka)"

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+Stroom uses Apache Kafka more extensively for quueing processing tasks and is capable of exposing the use of Kafka Streams for performing certain analytics.
+
+## Query fusion
+Stroom allows multiple data sources to be queried at the same time and the results of the queries to be fused. This might be for fusing data from multiple search indexes, e.g. events and annotations, or to effectively decorate results with additional data at search time.
+
+## Search result storage
```
Comment (gcdev373) [inline ROADMAP.md]: Great to do this.  But performance might be impacted.
Maybe consider using the disk backed memory store used for reference data for this job too?

File: ROADMAP.md (hunk 1 above)
Comment (gcdev373) [reply ROADMAP.md]: The second sentence I mean "In addition..."

File: ROADMAP.md (hunk 3 above)
Comment (stroomdev66) [reply ROADMAP.md]: Yes that is being considered plus fusing data should not impact the immediacy of returned results but instead be performed as a background decoration, i.e. an incremental update.

File: ROADMAP.md (hunk 2 above)
Comment (stroomdev66) [reply ROADMAP.md]: Agreed. The intent at least at this stage is not to make processing rely heavily on Kafka but instead add additional support for ingest from Kafka topics.

File: ROADMAP.md (hunk 1 above)
Comment (stroomdev66) [reply ROADMAP.md]: Which?

**Pull request merged**